File: app/src/main/java/com/neome/api/meta/base/dto/DefnStudioVarIdParagraphEditor.java
// neome.ai API - do not change
//

package com.neome.api.meta.base.dto;

import com.neome.api.meta.base.Types.EnumStudioVarKind;
import com.neome.api.meta.base.Types.MetaIdComposite;
import com.neome.api.meta.base.Types.MetaIdField;
import com.neome.api.meta.base.Types.MetaIdForm;
import com.neome.api.meta.base.Types.MetaIdPlugin;

import org.jetbrains.annotations.Nullable;

@SuppressWarnings("unused")
public class DefnStudioVarIdParagraphEditor extends DefnField
{
  @Nullable
  public EnumStudioVarKind[] argBinderFilterVarKindSet;

  public MetaIdForm argBinderFormId;

  @Nullable
  public MetaIdComposite[] compositeIdSet;

  @Nullable
  public MetaIdField[] excludeFieldIdSet;

  @Nullable
  public EnumStudioVarKind[] filterVarKindSet;

  @Nullable
  public MetaIdPlugin pluginId;

  @Nullable
  public Boolean required;

  @Nullable
  public Boolean showAsEdit;
}

File: app/src/main/java/com/neome/api/meta/base/dto/DefnStudioVarIdParagraphEditor.kt
// neome.ai API - do not change
//
@file:Suppress("unused", "MemberVisibilityCanBePrivate", "CanBeParameter", "PropertyName")


package com.neome.api.meta.base.dto

import com.neome.api.meta.base.Types.EnumStudioVarKind
import com.neome.api.meta.base.Types.MetaIdComposite
import com.neome.api.meta.base.Types.MetaIdField
import com.neome.api.meta.base.Types.MetaIdForm
import com.neome.api.meta.base.Types.MetaIdPlugin

interface DefnStudioVarIdParagraphEditor : DefnField {
    val argBinderFilterVarKindSet: List<EnumStudioVarKind>?
    val argBinderFormId: MetaIdForm
    val compositeIdSet: List<MetaIdComposite>?
    val excludeFieldIdSet: List<MetaIdField>?
    val filterVarKindSet: List<EnumStudioVarKind>?
    val pluginId: MetaIdPlugin?
    val required: Boolean?
    val showAsEdit: Boolean?
}
